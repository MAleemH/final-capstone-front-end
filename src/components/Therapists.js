/* eslint-disable no-unused-expressions */
/* eslint-disable no-undef */
import React from 'react';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import '../css/Therapist.css';
import facebookImg from '../img/facebook.png';
import twitterImg from '../img/twitter.png';
import linkedinImg from '../img/linkedin.png';
import trashImg from '../img/trash.png';
import { deleteTherapist, fetchTherapists } from '../redux/Therapy/therapySlice';
import { getLocalUser } from './localStore';

function Therapists(props) {
  const myUse = getLocalUser() || [];
  const myUser = myUse?.user;
  const { myTherapists } = props;
  const dispatch = useDispatch();

  const handleDeleteTherapist = async (e, objId) => {
    e.preventDefault();
    await dispatch(deleteTherapist(objId));
    dispatch(fetchTherapists());
  };

  return (

    <div className="home_grids">

      {myTherapists?.map((therapist) => (
        <div key={therapist.id} className="home_grids_box">
          <Link to={`/therapistdetails/${therapist.id}`} className="td_none">
            <article>
              <figure className="therapy_figure">
                <img src={therapist.photo} alt="" />
              </figure>

              <h4>{therapist.name}</h4>

              <p>{therapist.specialization}</p>

              <div className="home_page_socials">
                <figure className="home_page_social_icons">
                  <Link to="/" className="td_none">
                    <img src={facebookImg} alt="" />
                  </Link>
                </figure>

                <figure className="home_page_social_icons">
                  <Link to="/" className="td_none">
                    <img src={twitterImg} alt="" />
                  </Link>
                </figure>

                <figure className="home_page_social_icons">
                  <Link to="/" className="td_none">
                    <img src={linkedinImg} alt="" />
                  </Link>
                </figure>
              </div>

              {myUser?.role === 'admin' && (
                <div className="home_page_socials">
                  <figure className="home_page_social_icons">
                    <button type="button" onClick={(e) => handleDeleteTherapist(e, therapist.id)}>
                      <img src={trashImg} alt="" />
                    </button>
                  </figure>
                </div>
              )}

            </article>
          </Link>
        </div>
      ))}

    </div>
  );
}

Therapists.propTypes = {
  myTherapists: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.object)).isRequired,
};

export default Therapists;
