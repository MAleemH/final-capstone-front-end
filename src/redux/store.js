import { configureStore } from '@reduxjs/toolkit';
import therapyReducer from './Therapy/therapySlice';
import userReducer from './User/userSlice';

const store = configureStore({
  reducer: {
    therapy: therapyReducer,
    user: userReducer,
  },
});

export default store;
