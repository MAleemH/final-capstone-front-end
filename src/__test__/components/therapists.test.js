import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Therapists from '../../components/Therapists';

it('Chart component renders correctly', () => {
  const myTherapists = store.getState().therapist.therapists;
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Therapists myTherapists={myTherapists} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
