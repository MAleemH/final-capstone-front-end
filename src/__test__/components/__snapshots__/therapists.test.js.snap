// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Therapist component renders correctly 1`] = `
<div
  className="home_grids"
/>
`;
